src/main/java/uk/gov/justice/digital/domain/DomainExecutor.java

```diff
@@ -127,16 +127,21 @@ public Dataset<Row> getAllSourcesForTable(final String sourcePath, final String
 
 
     protected Dataset<Row> applyViolations(final Dataset<Row> dataFrame, final List<ViolationDefinition> violations) {
-        Dataset<Row> violationsDataFrame = dataFrame;
+        Dataset<Row> sourceDataframe = dataFrame;
         for (final ViolationDefinition violation : violations) {
-            final Dataset<Row> df_violations = violationsDataFrame.where("not(" + violation.getCheck() + ")").toDF();
-            if (!df_violations.isEmpty()) {
-                TableInfo info = TableInfo.create(targetRootPath, violation.getLocation(), violation.getName());
-                saveViolations(info, df_violations);
-                violationsDataFrame = violationsDataFrame.except(df_violations);
+            final String applyCondition = violation.getCheck();
```

Comment:
Minor - don't forget that lombok gives you a `val` keyword that means you can just do `val applyCondition = ...`